### ISSUE TITLE
bug: batch query on source failed when the format is UPSERT_JSON

### ISSUE THREAD

**Issue by neverchanje** [risingwavelabs/risingwave#9771]
### Describe the bug

Since UPSERT_JSON is unhandled, it will reach the unreachable branch and raise a panic.
https://github.com/risingwavelabs/risingwave/blob/main/src/batch/src/executor/source.rs#L81

### To Reproduce

```
CREATE SOURCE city (
    "id" int,
    "name" varchar,
    "email_address" varchar,
    "credit_card" varchar,
    "city" varchar,
) WITH (
    connector = 'kafka',
    properties.bootstrap.server = '127.0.0.1:29092',
    topic = 'city'
) ROW FORMAT UPSERT_JSON;

SELECT * FROM city;
```

UPSERT_AVRO is unhandled as well.

**Comment by wugouzi** [comment]
Fixed in #9833. But we need to disable this kind of query because upsert needs delete op, which is not permitted on source and we can only do such query after creating materialized view. Example:

```
CREATE TABLE city (
    "id" int primary key,
    "name" varchar,
    "email_address" varchar,
    "credit_card" varchar,
    "city" varchar,
) WITH (
    connector = 'kafka',
    properties.bootstrap.server = '127.0.0.1:29092',
    topic = 'city'
) ROW FORMAT UPSERT_JSON;

CREATE MATERIALIZED VIEW city_view AS
SELECT * FROM city;

SELECT * FROM city_view;
```

In addition, if we don't specify the primary key here, RisingWave will panic at 
'called `Option::unwrap()` on a `None` value', `src/stream/src/executor/row_id_gen.rs:84:65`. I will report it later.

**Comment by tabVersion** [comment]
after a discussion with @liurenjie1024 , batch query only accepts `Op::Insert`. we may ban CDC formats (debezium/maxwell/upsert) on batch query
